fml/raster_thread_merger.cc

```diff
@@ -80,34 +74,32 @@ bool RasterThreadMerger::IsOnRasterizingThread() const {
 }
 
 void RasterThreadMerger::ExtendLeaseTo(size_t lease_term) {
+  FML_DCHECK(lease_term > 0) << "lease_term should be positive.";
   if (TaskQueuesAreSame()) {
     return;
   }
-  std::scoped_lock lock(lease_term_mutex_);
-  FML_DCHECK(IsMergedUnSafe()) << "lease_term should be positive.";
-  if (lease_term_ != kLeaseNotSet &&
-      static_cast<int>(lease_term) > lease_term_) {
-    lease_term_ = lease_term;
-  }
+  std::scoped_lock lock(mutex_);
+  shared_merger_impl_->ExtendLeaseTo(lease_term);
```

Comment:
While this is threadsafe, I'm not sure of the ramifications of having potentially 2 RasterThreadMerger's calling into this at the same time overwriting the shared value, @iskakaushik ?

Reply:
See below.